### PR TITLE
Unittest/arrow to row data converters

### PR DESCRIPTION
**Pull request by Sridhar-Durgam2022**
ArrowToRowDataConverters unit test cases

### CODE REVIEW

File: src/test/java/com/google/cloud/flink/bigquery/unit/ArrowToRowDataConvertersTest.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import org.apache.flink.table.types.logical.RowType;
+import org.apache.flink.table.types.logical.RowType.RowField;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class ArrowToRowDataConvertersTest {
+
+  @Test
+  public void createRowConverterTest() {
+
```
Comment (davidrabinowitz) [inline src/test/java/com/google/cloud/flink/bigquery/unit/ArrowToRowDataConvertersTest.java:33]: Please test all types

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]
Please test all types and address sonatype comments

File: src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+        instance = null;
+      } finally {
+        instantiated = true;
+      }
+      return instance;
+    }
+
+    public long convertDate(Object object) {
+      final org.joda.time.LocalDate value = (org.joda.time.LocalDate) object;
+      return value.toDate().getTime();
```
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java:312]: @davidrabinowitz , We can discuss on this as Spark connector using the same

File: src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+      return value.get(DateTimeFieldType.millisOfDay());
+    }
+
+    public long convertTimestamp(Object object) {
+      long dateTime = 0;
+      if (!(object.toString().length() == 26)) {
+        final DateTime value =
+            DateTime.parse(object.toString(), ISODateTimeFormat.dateTimeParser().withZoneUTC());
+
+        dateTime = value.toDate().getTime();
```
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java:326]: @davidrabinowitz , We can discuss on this as Spark connector using the same

File: src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+      } else if (object.toString().length() == 26) {
+        DateTime value =
+            DateTime.parse(object.toString(), ISODateTimeFormat.dateTimeParser().withZoneUTC());
+        dateTime = value.toDate().getTime();
```
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java:330]: @davidrabinowitz , We can discuss on this as Spark connector using the same

File: pom.xml [hunk 5]
```diff
@@ -236,6 +237,16 @@
             <artifactId>jsqlparser</artifactId>
             <version>${jsqlparser.version}</version>
         </dependency>
+        <dependency>
+           <groupId>joda-time</groupId>
+           <artifactId>joda-time</artifactId>
+           <version>${joda.version}</version>
+        </dependency>
+        <dependency>
```
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply pom.xml:246]: changed to latest version

File: src/test/java/com/google/cloud/flink/bigquery/unit/ArrowToRowDataConvertersTest.java (hunk 1 above)
Comment (Sridhar-Durgam2022) [reply src/test/java/com/google/cloud/flink/bigquery/unit/ArrowToRowDataConvertersTest.java]: Tested with all types

File: src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java:312]: @davidrabinowitz , We can discuss on this as Spark connector using the same

File: src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java:330]: @davidrabinowitz , We can discuss on this as Spark connector using the same

File: src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java:326]: @davidrabinowitz , We can discuss on this as Spark connector using the same

File: pom.xml (hunk 5 above)
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply pom.xml:246]: updated to latest version

File: src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java [hunk 6]
```diff
@@ -0,0 +1,338 @@
...
+      return value.get(DateTimeFieldType.millisOfDay());
+    }
+
+    public long convertTimestamp(Object object) {
+      long dateTime = 0;
+      if (!(object.toString().length() == 26)) {
+        final DateTime value =
+            DateTime.parse(object.toString(), ISODateTimeFormat.dateTimeParser().withZoneUTC());
+
+        dateTime = value.toDate().getTime();
```
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java:326]: @davidrabinowitz , We can discuss on this as Spark connector using the same

File: src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java [hunk 7]
```diff
@@ -0,0 +1,338 @@
...
+      } else if (object.toString().length() == 26) {
+        DateTime value =
+            DateTime.parse(object.toString(), ISODateTimeFormat.dateTimeParser().withZoneUTC());
+        dateTime = value.toDate().getTime();
```
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java:330]: @davidrabinowitz , We can discuss on this as Spark connector using the same

File: src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java [hunk 8]
```diff
@@ -0,0 +1,338 @@
...
+        instance = null;
+      } finally {
+        instantiated = true;
+      }
+      return instance;
+    }
+
+    public long convertDate(Object object) {
+      final org.joda.time.LocalDate value = (org.joda.time.LocalDate) object;
+      return value.toDate().getTime();
```
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java:312]: @davidrabinowitz , We can discuss on this as Spark connector using the same

File: src/main/java/com/google/cloud/flink/bigquery/ArrowRowDataDeserializationSchema.java [hunk 9]
```diff
@@ -37,16 +41,45 @@
...
+        MessageSerializer.deserializeSchema(
+                new ReadChannel(
+                    new ByteArrayReadableSeekableByteChannel(
+                        arrowSchemafromReadSession.getSerializedSchema().toByteArray())))
+            .getFields().stream()
+            .forEach(
+                field -> {
+                  readSessionFieldNames.add(field.getName());
+                });
+        arrowSchema =
```
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/ArrowRowDataDeserializationSchema.java]: Addressed

File: src/main/java/com/google/cloud/flink/bigquery/BigQuerySinkFunction.java [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+
+  public BigQuerySinkFunction(
+      org.apache.flink.table.api.Table sourceResultTable,
+      String projectId,
+      String dataset,
+      String table)
+      throws JSQLParserException {
+    BigQueryDirectDataWriterContext writeContextObj =
+        new BigQueryDirectDataWriterContext(sourceResultTable, projectId, dataset, table);
+    writeContext = writeContextObj;
```
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/BigQuerySinkFunction.java]: Class is not required

File: src/test/java/com/google/cloud/flink/bigquery/ArrowToRowDataConvertersTest.java [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+            "date_dataType",
+            "decimal_dataType",
+            "double_dataType",
+            "row_dataType",
+            "timestamp_dataType",
+            "time_dataType");
+
+    ArrowToRowDataConverter createRowConverter =
+        ArrowToRowDataConverters.createRowConverter(rowType, readSessionFieldNames, selectedFields);
+    Assert.assertNotNull(createRowConverter);
```
Comment (VinayLondhe14) [inline src/test/java/com/google/cloud/flink/bigquery/ArrowToRowDataConvertersTest.java:79]: Just testing that the `createRowConverter` is not null isn't sufficient here. Where are tests for other methods in the `ArrowToRowDataConvertersTest` class

Comment (Sridhar-Durgam2022) [reply src/test/java/com/google/cloud/flink/bigquery/ArrowToRowDataConvertersTest.java:79]: @VinayLondhe14 , All other methods are static private which are invoked at run time conversion part in the process of deserialization (ie;ArrowRowDataDeserializationSchema -> rowData = (RowData) runtimeConverter.convert(root);)
So we can not test those methods directly.

**Pull request closed without merging**